### PR TITLE
Add support for multiple communities

### LINKED ISSUE #22: Manage multiple communities
Currently it is only possible to choose one community to report results in. It makes sense to be able to add multiple communities on the settings page:
- Instead of CommunityName/PlayerName, display a list of communities (as "CommunityName (playerName)" in the list)
- Limit the number of communities to 3
- "Add"-button in the toolbar to open the dialog for adding a Community/Player pair to the list
- "Delete"-button next to each item (possibly a [swipe-view](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/swipeview) for managing the list)
- It should not be possible to delete all communities
- On first run (while the list is still empty, the "OpponentList"-tab should be disabled, just as in the current implementation
- One tab per community at the top of the "OpponentList"-page (see [Fabimals](https://github.com/fsprojects/Fabulous/blob/master/Fabulous.XamarinForms/samples/Shell/Fabimals/Fabimals/App.fs#L163) sample)

### PR DESCRIPTION
**Pull request by MargaretKrutikova**
This PR fixes #22 by adding ability to have multiple communities with functionality to add and remove communities.

I added `getSelectedCommunity` helper in `AppSettings` that just fetched first community from he list when you need to fetch players and report results, since we haven't implemented tabs yet, but that's a pretty dirty solution for now. Do you have a better suggestion in mind? 😄

### CODE REVIEW

**Comment by Yakimych** [comment]
Looks like there are merge conflicts... Not sure why though - looks like all 4 commits in this PR came on top of the current master-branch. Any ideas?

Edit: never mind, it's fine if I choose "Squash and Merge", but only complains when I choose "Rebase and Merge".

File: src/Relogify/Relogify.fs [hunk 1]
```diff
@@ -177,12 +182,14 @@ module App =
...
 
+        let communities = (getApplicationSettings ()).Communities
+
         View.NavigationPage(
             ref = navigationPageRef,
             barBackgroundColor = navigationPrimaryColor,
             barTextColor = Color.White,
             popped = (fun _ -> dispatch PopPage),
-
+
```
Comment (Yakimych) [inline src/Relogify/Relogify.fs]: Visual studio is not great at trimming trailing whitespace...

**Comment by MargaretKrutikova** [comment]
I don't see any conflicts, everything is green here 🟢

**Review by Yakimych — CHANGES_REQUESTED** [review]
Found some issues after a quick test:
- On first run I end up on the "Players" page with an error message: "Application settings are not set" and a "Try again" button. This should never happen, the "Players"-tab should be invisible (disabled), and we should be on the "Settings" page and forced to add at least one community.
- I should be able to Cancel the dialog if at least one community already exists. Currently there is no "Cancel" button even if I've already added a community previously.

Bonus: might be worth writing a unit test for the second issue before implementing. It should be relatively straightforward with the state-machine-like setup we have.

File: src/Relogify/ApplicationSettings.fs [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
 
-let CommunityNameStorageKey = "community_name"
-let PlayerNameStorageKey = "player_name"
+type ApplicationSettings =
+    { Communities: Community list }
 
-let getStringValueOrNone key =
-    let found, property = Application.Current.Properties.TryGetValue(key)
-    if found then Some(property.ToString()) else None
+let ApplicationSettinsStorageKey = "app_settings_key"
```
Comment (Yakimych) [inline src/Relogify/ApplicationSettings.fs:13]: ApplicationSettin**G**sStorageKey

**Comment by Yakimych** [comment]
@MargaretKrutikova Could you re-open this PR? I pushed a commit from here to master by mistake, and GitHub closed the PR automatically.

**Pull request merged**